File: data-science/research/fincoach-mvp/ds-service/movements/application/classify_movement_use_case.py
from typing import Dict, Any
from movements.domain.model_classifier import ModelClassifier

class ClassifyMovementUseCase:
    @staticmethod
    def execute(
        description: str,
        amount: float,
        direction: str = "salida",
        note: str = ""
    ) -> Dict[str, Any]:
        """Ejecuta la orquestación del caso de uso de clasificación de movimiento."""
        return ModelClassifier.classify(
            description=description,
            amount=amount,
            direction=direction,
            note=note
        )

File: backend-SpringBoot/src/main/java/com/g9latam/team14/movement/domain/ports/inbound/ClassifyMovementUseCase.java
package com.g9latam.team14.movement.domain.ports.inbound;
import com.g9latam.team14.movement.domain.model.AiClassification;
public interface ClassifyMovementUseCase {
    AiClassification classifyMovement(String description, double amount, String direction, String note);
}
